lib/clear.js

```diff
@@ -0,0 +1,169 @@
+        .catch(err => console.log(err))
+        }
+    ).catch(err => console.log(err));
+}
+
+const deleteShippingMethodsAndZones = () => {
+    client.execute({
+        uri: shippingMethodService.perPage(500).build(),
+        method: 'GET',
+    }).then(result => {
+        return Promise.all(
+            result.body.results.map((element) => {
+                return client.execute({
+                    uri: shippingMethodService.byId(element.id).withVersion(element.version).build(),
+                    method: 'DELETE',
+                })
+            })
+        ).then(() => {
+            console.log('All shipping methods are succesfully deleted')
+            client.execute({
+                uri: zonesService.perPage(500).build(),
+                method: 'GET',
+            }).then(result => {
+                return Promise.all(
+                    result.body.results.map((element) => {
+                        return client.execute({
+                            uri: zonesService.byId(element.id).withVersion(element.version).build(),
+                            method: 'DELETE',
+                        })
+                    })
+                ).then(console.log('All zones are succesfully deleted'))
+                .catch(err => console.log(err))
+                }
+            ).catch(err => console.log(err));
+        })
+        .catch(err => console.log(err))
+        }
+    ).catch(err => console.log(err));
+}
+
+const deleteProducts = () => { 
+    client.execute({
+        uri: productsService.perPage(500).build(),
+        method: 'GET',
+    }).then(result => {
+        if (result.body.count > 0) {
+            console.log('Deleting ' + result.body.count + ' product(s) ..');
+             Promise.all(
+                result.body.results.map((element) => {
+                        return client.execute({
+                            uri: productsService.byId(element.id).build(),
+                            method: 'POST',
+                            body: {
+                                version: element.version,
+                                actions: [{
+                                    action: 'unpublish'
+                                  }]
+                            }
+                        }).then(result => {
+                            return client.execute({
+                                uri: productsService.byId(result.body.id).withVersion(result.body.version).build(),
+                                method: 'DELETE',
+                            })
+                        }  
+                        )
+                        .catch(err => console.log(err))
+                })
+            ).then(() => {
+                deleteProducts()
+            })
+            .catch(err => console.log(err))
+        } else {
+            console.log('All products are successfully deleted')
+        };
+    })
+    .catch(err => console.log(err));
+}
+
+
+deleteCategories();
```

Comment:
I don't find it super useful to provide just one call that deletes _everything_.  If I really want everything deleted I'll rather delete the whole project, create a new one and run the importer from scratch.  That's safer and faster. 

But there are typical use cases for deleting, that's typically either clearing and recreating the product catalog only (i.e.  product types, products and categories) 

or, more relevant in a demo or POC, clear the transactional data only but keep your configuration and product catalog.  You want to clear customers, carts, orders, payments, shopping lists.  While setting up the demo or POC you create test and dummy names and orders. For the actual demo you want to start clean or have well named custmer and order entries.  

Another thing you may want to purge as a block of topics to recreate them from scratch is discount codes, cart discounts and product discounts.

Reply:
I totally agree with you on that, we can discuss it with @sshibani